otterai/indexer.py

```diff
@@ -42,59 +45,78 @@ def index_codebase(root_dir: str) -> Dict[str, List[str]]:
+        '*.pyc', '__pycache__/*', '.git/*', '.github/*', 'node_modules/*',
+        '*.min.js', '*.min.css', '*.map', '*.lock', '*.sum',
+        'dist/*', 'build/*', '.env*', '*.log',
+        '*venv/*', '*.venv/*', '*.venv', 'venv/*', 'venv', '*.venv',
+        '*.pyc', '__pycache__/*', '.git/*', '.github/*', 'node_modules/*',
+    ]
     
     for root, _, files in os.walk(root_dir):
         for file in files:
             file_path = os.path.join(root, file)
             rel_path = os.path.relpath(file_path, root_dir)
             
             if should_ignore_file(rel_path):
+                print(f"Ignoring file: {file} because it matches ignore patterns.")
                 continue
                 
             file_type = get_file_type(rel_path) or 'other'
             index[file_type].append(rel_path)
     
     return index
 
-def analyze_project_structure(index: Dict[str, List[str]], repo_root: str) -> str:
+async def analyze_project_structure(index: Dict[str, List[str]], repo_root: str) -> str:
     """Generate a high-level analysis of the project structure."""
-    llm = ChatOpenAI(
-        model=os.getenv('INPUT_MODEL', 'gpt-4-turbo-preview'),
-        api_key=os.getenv('INPUT_OPENAI_API_KEY'),
-        base_url=os.getenv('INPUT_OPENAI_BASE_URL', 'https://api.openai.com/v1'),
-        temperature=0.1
-    )
+    llm_client = LLMClient()
+    llm = llm_client.get_client()
     
     prompt = ChatPromptTemplate.from_messages([
         ("system", """You are a technical architect analyzing a codebase structure.
-        Create a concise but comprehensive overview of the project structure and guidelines.
-        Focus on:
-        1. Project organization and architecture
-        2. Key components and their relationships
-        3. Coding standards and patterns observed
-        4. Important dependencies and configurations
-        5. Testing approach
-        
-        Keep the response focused and actionable for code review purposes."""),
+Create a concise but comprehensive overview of the project structure and guidelines.
+Focus on:
+1. Project organization and architecture
+2. Key components and their relationships
+3. Coding standards and patterns observed
+4. Important dependencies and configurations
+5. Testing approach
+ 
+Steps:
+1. Read the codebase structure
+2. Read the key files' content
+3. Generate the analysis
+4. Strictly check which language is used in the codebase; do not mention any other language if not specified.
+
+Do not use any other language than English. Do not leak any sensitive information.
+
+Keep the response focused and actionable for code review purposes."""),
         ("human", """Here's the codebase structure:
-        
-        {index_summary}
-        
-        Key files content:
-        {key_files_content}
-        """)
+
+{index_summary}
+
+Key files content:
+{key_files_content}
+""")
     ])
     
-    # Read content of key files
+    # Read content of key files asynchronously
```

Comment:
📁 The 'read_file' function is used to read the content of key files asynchronously. Ensure that all file reads are properly awaited to prevent blocking.